QueueMessageSender/QueueMessageSender/Logic/RmqMessageSender.cs

```diff
@@ -10,60 +10,137 @@ namespace QueueMessageSender.Logic
-                    }
-                }
+                return _connectionLazy.Value;
             }
         }
+        #endregion
 
-        private static void CreateChannel(object status)
+        #region Channel field and property
+        private Lazy<IModel> _channelLazy = new Lazy<IModel>(() =>
         {
-            _channel.Close();
-            _channel = _connection.CreateModel();
+            return null;
+        });
+        private IModel Channel
+        {
+            get
+            {
+                return _channelLazy.Value;
+            }
         }
+        #endregion
 
-        public void SendMessage(DepartureData data)
+        #region Channel field and property
+        private Lazy<List<string>> _namesExchangeLazy = new Lazy<List<string>>(() =>
         {
-            if (_factory == null || _connection == null || _channel == null)
+            return new List<string>();
+        });
+        private List<string> NamesExchange
+        {
+            get
             {
-                CreateConnection();
+                return _namesExchangeLazy.Value;
             }
+        }
+        #endregion
+
+        private void CreateConnection()
+        {
+            _factoryLazy = new Lazy<ConnectionFactory>(() =>
+            {
+                ConnectionFactory factory = new ConnectionFactory() { HostName = "localhost" };
+                factory.AutomaticRecoveryEnabled = true;
+
+                return factory;
+            });
+
+            _connectionLazy = new Lazy<IConnection>(() =>
+            {
+                return Factory.CreateConnection();
+            });
 
-            if (!_namesExchange.Contains(data.NameExchange))
+            _channelLazy = new Lazy<IModel>(() =>
             {
-                _channel.ExchangeDeclare(exchange: data.NameExchange, type: ExchangeType.Fanout);
-                _namesExchange.Add(data.NameExchange);
+                return Connection.CreateModel();
+            });
+
+            Timer timer = new Timer(CreateChannel, 0, Convert.ToInt32(Channel.ContinuationTimeout.TotalMilliseconds), Convert.ToInt32(Channel.ContinuationTimeout.TotalMilliseconds));
+        }
+
+        private void CreateChannel(object status)
+        {
+            Channel.Close();
+            _channelLazy = new Lazy<IModel>(() =>
+            {
+                return Connection.CreateModel();
+            });
+        }
+
+        private void VerifyExchangeCreation(string nameExchange) 
+        {
+            if (!NamesExchange.Contains(nameExchange))
+            {
+                try
+                {
+                    Channel.ExchangeDeclare(exchange: nameExchange, type: ExchangeType.Fanout);
+                    NamesExchange.Add(nameExchange);
+                }
+                catch (RabbitMQ.Client.Exceptions.AlreadyClosedException)
+                {
+                    CreateConnection();
```

Comment:
1. Please think on correct processing of requests. Message should be send even in case if connection was re-created after AlreadyClosedException. This is related to all try/catch sections in this class.
2. Can't connection.closed event be used to check when connection is closed?